keps/sig-network/1111-dynamically-re-configure-lb-node-set/README.md

```diff
@@ -0,0 +1,681 @@
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+#### Prerequisite testing updates
+
+#### Unit tests
+
+Components in the architecture presented above are already existing (and
+supposedly are already unit tested): kube-proxy's healthz server, kube-proxy's
+"ETP=local" server. We would need to add tests which validate the change in
+behaviour for opening up additional ports for ETP=cluster services and validate
+that the difference in HC logic between ETP services is captured correctly.
+
+#### Integration tests
+
+Kube-proxy is mostly tested via unit tests and e2e, not integration, and this is
+not expected to change.
+
+#### e2e tests
+
+Kubernetes in general needs to extended its load balancing test suite with
+disruption tests, this might be the right effort we need to get that ball
+rolling. Testing would include: validation that HC probe's return a respond
+corresponding to the "state of the Kubernetes cluster world" during different
+disruption scenarios.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature flag for changing the API for ETP=cluster services and
+  include `healthCheckNodePort` for them. This will help testing and graduation
+  to beta and stable.  
+- Changes to data plane components get implemented (Kube-proxy)
+- Changes to control plane components get implemented, but depends on the
+  feature flag being enabled or not.
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Documentation is changed for ETP=cluster services to highlight the change with
+  regards to `healthCheckNodePort`.
+- Testing of Kubernetes with the feature flag enabled is green and reports no
+  major incidents.
+
+#### GA
+
+- Version skew delta has passed.
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+### Version Skew Strategy
+
+The possible version skew between the control and data plane in a Kubernetes
+cluster and the fact that this proposal depends on having the data plane
+implement an HC which the control plane can target, makes the implementation
+plan tricky. In a world where this KEP has been agreed upon, we would need to
+begin by implementing the HC in the data plane, allow it to soak until the
```

Comment:
can you clarify what do you mean by "data plane" and "control plane" concepts?
I may have an idea of what you try to mean, but this concept is heavily tied to routing platforms and can be misleading, mainly because kubernetes doesn't have this separation in terms of network, at least not physically

Reply:
Re-reading the paragraph I realize I said something wrong, the control plane does not "target" the HC, it programs the HC and the provisioned load balancer "targets"/uses it.

By control plane I mean: set of nodes where user workloads do not run (API server & friends). By data plane I mean the inverse.

Reply:
I see, I think we can add that explanation to make it easier to understand, as "implementing the HC in all nodes that runs workloads" or something similar,